bots/mirror/src/main/java/org/openjdk/skara/bots/mirror/MirrorBotFactory.java

```diff
@@ -79,10 +79,23 @@ public List<Bot> create(BotConfiguration configuration) {
                 branchPatterns = List.of();
             }
 
-            var includeTags = branchPatterns.isEmpty() || (repo.contains("tags") && repo.get("tags").asBoolean());
+            var includeTags = branchPatterns.isEmpty();
+            var onlyTags = false;
+            if (repo.contains("tags")) {
+                includeTags = repo.get("tags").contains("include") &&
+                              repo.get("tags").get("include").asBoolean();
+                onlyTags = repo.get("tags").contains("only") &&
+                           repo.get("tags").get("only").asBoolean();
+            }
+            if (onlyTags && !includeTags) {
+                throw new IllegalStateException("Must include tags if only tags are mirrored");
+            }
+            if (onlyTags && !branchPatterns.isEmpty()) {
+                throw new IllegalStateException("Branches cannot be mirrored when only tags are mirrored");
+            }
```

Comment:
Originally, if `branchPatterns.isEmpty()`, `includeTags` would always be true. But now, it is possible that when `branchPatterns.isEmpty()`, `includeTags` is false.
So I think we could also throw exception when `branchPatterns.isEmpty() && !includeTags`. It is illegal.

Reply:
Thanks @zhaosongzs , I reworked this a bit in the latest version and fixed this as well